lib/src/validationEngine.ts

```diff
@@ -1,62 +1,64 @@
+import { Promise } from 'core-js';
 import {
   FormValidationResult,
   FieldValidationResult,
   FieldValidation,
+  FieldValidationFunction,
   ValidationResult,
   FormValidationFunction,
+  ValidationFilters,
 } from "./entities";
 import { consts } from './consts';
 import { validationsDispatcher } from './validationsDispatcher';
 import { validationsResultBuilder } from './validationsResultBuilder';
 import { fieldValidationEventFilter } from './fieldValidationEventFilter';
-import { entitiesMapper } from './entitiesMapper';
 
 export interface IValidationEngine {
   isFormDirty(): boolean;
   isFormPristine(): boolean;
+  isValidationInProgress(): boolean;
   validateForm(vm: any): Promise<FormValidationResult>;
-  triggerFieldValidation(vm: any, key: string, value: any, filter?: any): Promise<FieldValidationResult>;
+  triggerFieldValidation(vm: any, key: string, value: any, filter?: ValidationFilters, customParams?: Object): Promise<FieldValidationResult>;
```

Comment:
There is a method called `validateSingleField` that is called by `triggerFieldValidation`. Could be a mess rename it `validateField` (that calls `validateSingleField`)?